soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,201 @@
+                        'after getting the ball to the center of the field, pass to the distracting robot')
+
+        self.add_transition(distraction.State.passing, 
+                        distraction.State.cross, lambda: 
+                        (self.has_subbehavior_with_name('distract pass') and 
+                            self.subbehavior_with_name('distract pass').is_done_running()) or
+
+                        (self.has_subbehavior_with_name('get close ball') and 
+                            self.subbehavior_with_name('get close ball').is_done_running()), 
+                        'have the distracrting robot capture the ball and go to passing to the striker')
+
+        self.add_transition(distraction.State.passing,
+                        distraction.State.shoot, lambda: 
+                        (self.has_subbehavior_with_name('striker pass') and self.subbehavior_with_name('striker pass').is_done_running() ) or 
+
+                        (self.has_subbehavior_with_name('striker get close ball') and 
+                            self.subbehavior_with_name('striker get close ball').is_done_running()) or
+
+                        (self.has_subbehavior_with_name('get close ball') and 
+                            self.subbehavior_with_name('get close ball').is_done_running()) or
+
+                        (self.has_subbehavior_with_name('distract pass') and 
+                            self.subbehavior_with_name('distract pass').is_done_running()),
+                        'if the ball is already close to the ball, then go from the striker captureing the ball to shooting the ball')
+
+        self.add_transition(distraction.State.cross,
+                        distraction.State.shoot, lambda: 
+                        self.subbehavior_with_name('pass to striker').is_done_running() or 
+
+                        (self.has_subbehavior_with_name('capture 3') and 
+                            self.subbehavior_with_name('capture 3').is_done_running()) , 
+                        'go from the striker receiving the pass to shooting the ball')
+
+        self.add_transition(distraction.State.shoot,
+                        distraction.State.passing, lambda: self.dont_shoot,
+                        "don't shoot if shoot chance is too low")
+
+        self.add_transition(distraction.State.shoot,
+                        distraction.State.setup, lambda: 
+                        self.subbehavior_with_name('shooting').is_done_running(), 
+                        'repeat')
+
+    def on_enter_setup(self):
+        self.remove_all_subbehaviors()
+        #capture ball and get striker and distractor in position
+        self.add_subbehavior(skills.capture.Capture(), 'capture', required = True)
+        
+        if not self.distracter_get_close_ball:
+            self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'distract moves', required = True)
```

Comment:
Some of these `add_subbehavior` calls are getting long, I would format the longer ones so each argument is on a different line. 

```
self.add_subbehavior(skills.move.Move(asdf),
                     'asdf',
                     required = True)
```